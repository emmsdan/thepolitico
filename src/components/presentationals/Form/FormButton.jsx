import React from 'react';
import PropTypes from 'prop-types';

const FormButton = props => {
  return (
    <div className="field">
      <button
        {...props}
        type={props.type}
        className={`button-bigger ${props.className}`}
      >
        {props.text}
      </button>
    </div>
  );
};

FormButton.propTypes = {
  text: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  className: PropTypes.string,
};

export default FormButton;
