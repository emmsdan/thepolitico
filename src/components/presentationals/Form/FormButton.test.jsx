import React from 'react';
import { shallow } from 'enzyme';

import FormButton from './FormButton';

describe('<FormButton />', () => {
  const button = {
    type: 'submit',
    className: 'bg-gray',
    text: 'Create an Account today',
  };

  it('should render correctly', () => {
    const component = shallow(<FormButton {...button} />);
    expect(component).toMatchSnapshot();
  });

  it('should display button text correctly', () => {
    const component = shallow(<FormButton {...button} />);
    expect(component).toMatchSnapshot();
    expect(component.find('.bg-gray').text()).toEqual(button.text);
  });
});
