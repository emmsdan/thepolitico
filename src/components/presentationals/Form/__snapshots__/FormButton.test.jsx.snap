// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormButton /> should display button text correctly 1`] = `
<div
  className="field"
>
  <button
    className="button-bigger bg-gray"
    text="Create an Account today"
    type="submit"
  >
    Create an Account today
  </button>
</div>
`;

exports[`<FormButton /> should render correctly 1`] = `
<div
  className="field"
>
  <button
    className="button-bigger bg-gray"
    text="Create an Account today"
    type="submit"
  >
    Create an Account today
  </button>
</div>
`;
